sdk/core/azure-core/src/main/java/com/azure/core/http/policy/RetryPolicy.java

```diff
@@ -70,10 +69,10 @@ public RetryPolicy(int maxRetries, Duration delayDuration) {
     private boolean shouldRetry(HttpResponse response, int tryCount) {
         int code = response.statusCode();
         return tryCount < maxRetries
-                && (code == HttpResponseStatus.REQUEST_TIMEOUT.code()
-                || (code >= HttpResponseStatus.INTERNAL_SERVER_ERROR.code()
-                && code != HttpResponseStatus.NOT_IMPLEMENTED.code()
-                && code != HttpResponseStatus.HTTP_VERSION_NOT_SUPPORTED.code()));
+                && (code == 408  // Request timeout
+                || (code >= 500) // Internal server error
```

Comment:
Parenthesis were changed here

Reply:
Good catch. I was being quite hacky. I'll fix this now so if you want to continue this PR tomorrow it is already resolved.